src/cargo/ops/cargo_add/mod.rs

```diff
@@ -559,6 +621,21 @@ fn get_latest_dependency(
     }
 }
 
+fn parse_rust_version(rust_version: impl AsRef<str>) -> (u64, u64, Option<u64>) {
+    // HACK: `rust-version` is a subset of the `VersionReq` syntax that only ever has one comparator
+    // with a required minor and optional patch, and uses no other features. If in the future this
+    // syntax is expanded, this code will need to be updated.
+
+    let version = semver::VersionReq::parse(rust_version.as_ref()).unwrap();
+    assert!(version.comparators.len() == 1);
+
+    let version = &version.comparators[0];
+    assert_eq!(version.op, semver::Op::Caret);
+    assert_eq!(version.pre, semver::Prerelease::EMPTY);
+
+    (version.major, version.minor.unwrap(), version.patch)
```

Comment:
Why is this `unwrap` here?  I've not seen anything in the existing code require a `minor` field and there isn't an associated comment + assertion

Reply:
From the [manifest spec](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field):
> The Rust version must be a bare version number with two or three components

I interpreted that to mean a major and a minor field are required. Looking around though, I don't really see that being validated anywhere, which is a bit odd.  Practically speaking, right now `rust-version = 1` would fail per the edition requirement, so I doubt there is any code using that in the wild.

The comment does mention it, but I suppose an assertion wouldn't hurt as well.

Reply:
Seems like a bug then. If you think so too, I can fix that in another PR.

Reply:
Changing a working case to a broken case requires a deeper analysis to determine if its a major breaking change (won't happen) or a minor breaking change (can happen).  Even if we do, we more so provide a warning, giving people an extensive period of time to adjust.

With that said, we should be mirroring here what is supported today (ie don't error or panic) and deferring anything more to an issue.